noir-projects/noir-contracts/contracts/test_contract/src/main.nr

```diff
@@ -262,23 +267,20 @@ contract Test {
         let header = context.get_header();
         let outgoing_viewer_ovpk_m = header.get_ovpk_m(&mut context, outgoing_viewer);
         let owner_ivpk_m = header.get_ivpk_m(&mut context, owner);
-        context.encrypt_and_emit_event(
-            5, // testing only - this should be a secret random value to salt the addr
-            1,
-            outgoing_viewer_ovpk_m,
-            owner_ivpk_m,
-            fields
-        );
+
+        let event = ExampleEvent { value0: fields[0], value1: fields[1], value2: fields[2], value3: fields[3], value4: fields[4] };
+
+        event.emit(encode_and_encrypt_event_with_keys(&mut context, 5, outgoing_viewer_ovpk_m, owner_ivpk_m));
```

Comment:
Think it would be useful if you mark `5` as randomness, just to make clear what it is instead of a magic number.